java/src/main/java/io/cucumber/java/ObjectFactoryLoader.java

```diff
@@ -2,45 +2,55 @@
 
 import cucumber.api.java.ObjectFactory;
 import io.cucumber.core.io.ClassFinder;
-import io.cucumber.core.exception.CucumberException;
-import io.cucumber.core.reflection.NoInstancesException;
-import io.cucumber.core.reflection.Reflections;
-import io.cucumber.core.reflection.TooManyInstancesException;
 
-import static java.util.Arrays.asList;
+import java.util.Iterator;
+import java.util.ServiceLoader;
 
 public class ObjectFactoryLoader {
+
+    private static final ServiceLoader<ObjectFactory> LOADER = ServiceLoader.load(ObjectFactory.class);
```

Comment:
The service loader is not safe for use by multiple concurrent threads. Runners are executed in parallel and each runner will create it's own backends with object factory. You can either make all fields and methods non-static or move this field into the method.